Caltech-IPAC/firefly
# Firefly-1495: sd parsing and extraction fixes

####  [Firefly-1495](https://jira.ipac.caltech.edu/browse/FIREFLY-1495): sd parsing and extraction fixes

 - Improve service descriptor parsing for the following complex case 
      - data has data product
      - table has non-datalink service descriptor 
      - table also has datalink service descriptor
 - Improve error handling
 - Improve SSA data recognition: recognize DataID.Title utype
 - Fix "Center Plot / Recenter" for large 1xN fits data
 - For 1xN data give a line extract option to extract whole line
 - fix color color picker close but
 - fix finder chart 3C button

#### IFE PR: https://github.com/IPAC-SW/irsa-ife/pull/329

#### Testing 
   - finder chart: https://firefly-1495-sd-parsing.irsakudev.ipac.caltech.edu/applications/finderchart
   - Three color button updated
   - close works on the color panel
- Firefly: https://fireflydev.ipac.caltech.edu/firefly-1495-sd-parsing/firefly
    - use file: [text.xml.zip](https://github.com/user-attachments/files/15571218/text.xml.zip)
    - Upload the file
         - besides the image (the 1st option), the second option will allow you to do a search, which will fail
         - the reset button should work
         - The third option should show an error message
         - The forth and fifth are related to the now found datalink service descriptor
    - line extract the file
        - the line extract should have a button to extract the while file since it is 1xN
   - Recenter should work correctly
